pytest_timeout.py

```diff
@@ -136,6 +136,25 @@ def pytest_report_header(config):
         ]
 
 
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    """Set report.timed_out after it is generated by pytest_runtest_makereport.
+
+    This method is a wrapper that sets timed_out if the call has a matching excinfo
+    value. Note that in that case report.failed and report.timed_out are true (both!).
+    """
+    r = yield
+    if not hasattr(r, "get_result"):
+        return
+    report = r.get_result()
+    timed_out = False
+    if hasattr(call.excinfo, "value"):
+        msg = getattr(call.excinfo.value, "msg", None)
+        if isinstance(msg, str) and msg.startswith("Timeout >"):
+            timed_out = True
```

Comment:
I don't really like this method of detecting whether a timeout occurred.  We have access to `item` here, and we have access to the same `item` in `timeout_setup()`.  It seems like we could set an attribute, say `item.pytest_timeout__timed_out`, to keep track of this which I think would be more robust.

Reply:
Ah yes, that would be probably better! That way originated in my attempt to implement this feature outside of `pytest-timeout`. Now I really should use that power of internals. I'll have a look at it, thanks for the feedback!

Reply:
Done. The tests still pass, although I don't currently have a test for the customization feature. I probably should add one that checks the reliability of this new, more efficient technique using the old method that you highlighted above.